dashboard/app/models/unit_group.rb

```diff
@@ -312,24 +309,19 @@ def self.has_any_course_experiments?(user)
     Experiment.any_enabled?(user: user, experiment_names: UnitGroupUnit.experiments)
   end
 
-  # Get the set of valid courses for the dropdown in our sections table.
-  def self.valid_courses_without_cache
-    UnitGroup.all.select(&:launched?)
-  end
-
   # Returns whether the course id is valid, even if it is not "stable" yet.
   # @param course_id [String] id of the course we're checking the validity of
   # @return [Boolean] Whether this is a valid course ID
   def self.valid_course_id?(course_id, user = nil)
-    valid_courses(user: user).any? {|unit_group| unit_group.id == course_id.to_i}
+    UnitGroup.valid_courses(user: user).any? {|unit_group| unit_group.id == course_id.to_i}
   end
 
   # @param user [User]
   # @returns [Boolean] Whether the user can assign this course.
   # Users should only be able to assign one of their valid courses.
   def assignable_for_user?(user)
     if user&.teacher?
-      UnitGroup.valid_course_id?(id)
+      UnitGroup.valid_course_id?(id, user)
```

Comment:
this looks like it removes the last caller of `valid_course_id?` which does not specify a user. what do you think about making `user` a required param to help avoid this kind of problem in the future?

Reply:
That seems like a good idea. I'll try updating.